### ISSUE TITLE
Improved alloc exec error message on failure to connect to Nomad

### ISSUE THREAD

**Issue by RussellRollins** [hashicorp/nomad#10786]
### Proposal
The Nomad error message when running `nomad alloc exec --job` and attempting to connect to a Nomad address that does not connect or does not serve the Nomad API should be more helpful, for instance, matching the `nomad alloc exec` error message without the job flag.

### Nomad version
```
$ nomad version
Nomad v1.0.3 (08741d9f2003ec26e44c72a2c0e27cdf0eadb6ee)
```

#### Actual Result
Try to connect to a service that doesn't serve the Nomad API or is an invalid port. For instance, example.com does not listen on 4646.

```
$ NOMAD_ADDR="https://example.com:4646" nomad alloc exec --job "example" /bin/sh
Error fetching allocations: job "example" doesn't exist or it has no allocations
```

While it is true that example.com does not have a job "example" or it has no allocations, it's not helpful. I'd expect to see an error message that includes information about the connection not working.

#### Desired Result

```
$ NOMAD_ADDR="https://example.com:4646" nomad alloc exec --job "example" /bin/sh
Error fetching allocations: Get "https://example.com:4646/v1/jobs?prefix=example": net/http: TLS handshake timeout
```

### Use-cases
When I work with `nomad alloc exec --job`, I may make a mistake in my `NOMAD_ADDR` or be working in an environment without a reliable connection to Nomad. I want the error messages to explain the situation to me.

### Attempted Solutions
If I see a confusing error from `nomad alloc exec --job` I check `nomad job status`, which provides helpful error messages. This is fine, but it's a bit annoying.

**Comment by tgross** [comment]
Hi @RussellRollins! This sparked my curiosity and so I took a quick look at it, and it turns out you discovered a bug in `nomad alloc fs` as well!

When you use `alloc exec` with the `-job` flag, we call a function [(ref `alloc_exec.go#L163`)](https://github.com/hashicorp/nomad/blob/v1.1.1/command/alloc_exec.go#L163)) that picks a random allocation, and that's also used by `nomad alloc fs`. But that function ignores the error it gets back [(ref `alloc_fs.go#L383`)](https://github.com/hashicorp/nomad/blob/v1.1.1/command/alloc_fs.go#L383) unless the response _also_ includes allocations. Will of course it never will if we got an error. Nice find!

Marking this accepted for the roadmap. But if you happen to know a little go, I'd also be happy to review a PR from you for it. Thanks for opening this issue!
